src/certificate.rs

```diff
@@ -50,21 +55,17 @@ pub enum CertificateError {
     #[error("Certificate does not contain a trusted root.")]
     NoTrustedRoot,
     #[error("Root trust did not pass verification: {0}")]
-    MalformedRoot(String),
+    MalformedRoot(TrustError),
     #[error("There is no `issued_by` public key in a certificate")]
     KeyInCertificateError,
     #[error("The certificate must have at least 1 trust")]
     CertificateLengthError,
+    #[error("Cannot convert trust number {0} from string: {1}")]
+    DecodeTrustError(usize, TrustError),
```

Comment:
do that everywhere plz
```suggestion
    DecodeTrustError(usize, #[source]TrustError),
```